tests/std/tests/GH_002769_handle_deque_block_pointers/test.cpp

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cstddef>
+#include <deque>
+#include <iterator>
+#include <memory>
+#include <type_traits>
+
+using namespace std;
+
+size_t fancy_counter{};
+
+template <class T>

```

Comment:
That is some pretty extensive work needed for a small test. 

I am wondering whether it would make sense to reuse that machinery and test all containers for internal memory leaks?